src/settings.ts

```diff
@@ -1,11 +1,23 @@
 import FullCalendarPlugin from "./main";
-import { App, DropdownComponent, Notice, PluginSettingTab, Setting, TFolder, Vault } from "obsidian";
-import { makeDefaultPartialCalendarSource, CalendarSource, FCError } from "./types";
+import {
+	App,
+	DropdownComponent,
+	Notice,
+	PluginSettingTab,
+	Setting,
+	TFolder,
+} from "obsidian";
+import {
+	isCalDAV,
+	makeDefaultPartialCalendarSource,
+	CalendarSource,
+	FCError,
+} from "./types";
 import { CalendarSettings } from "./components/CalendarSetting";
 import { AddCalendarSource } from "./components/AddCalendarSource";
 import { RemoteSource } from "./models/EventSource";
 import * as ReactDOM from "react-dom";
-import { createElement, useState } from "react";
+import { createElement } from "react";
```

Comment:
This is a nitpick, but namespacing helper functions with `React` is helpful for readability. `createElement` doesn't look react-specific.
```suggestion
import * as React from "react";
```